src/tests/unit/tests/background/initial-assessment-store-data-generator.test.ts

```diff
@@ -1,32 +1,60 @@
+import { InitialDataCreator } from '../../../../background/create-initial-assessment-test-data';
 import { InitialAssessmentStoreDataGenerator } from '../../../../background/initial-assessment-store-data-generator';
-import { IAssessmentStoreData } from '../../../../common/types/store-data/iassessment-result-data';
+import { IAssessmentData, IAssessmentStoreData } from '../../../../common/types/store-data/iassessment-result-data';
 import { VisualizationType } from '../../../../common/types/visualization-type';
+import { DictionaryStringTo } from '../../../../types/common-types';
 import { CreateTestAssessmentProvider } from '../../common/test-assessment-provider';
 
 describe('InitialAssessmentStoreDataGenerator.generateInitialState', () => {
     const assesssmentsProvider = CreateTestAssessmentProvider();
-    const generator = new InitialAssessmentStoreDataGenerator(assesssmentsProvider);
+    const assessments = assesssmentsProvider.all();
     const validTargetTab = { id: 1, url: 'url', title: 'title', appRefreshed: false };
-    const knownTestType = assesssmentsProvider.all()[0].type;
+    const knownTestType = assessments[0].type;
     const unknownTestType = -100 as VisualizationType;
-    const knownRequirementIds = flatMap(assesssmentsProvider.all(), test => test.steps.map(step => step.key));
+    const knownRequirementIds = flatMap(assessments, test => test.steps.map(step => step.key));
     const knownRequirement1 = knownRequirementIds[0];
     const unknownRequirement: string = 'unknown-requirement';
     let defaultState: IAssessmentStoreData;
+    let initialDataCreatorMock: IMock<InitialDataCreator>;
+    let generator: InitialAssessmentStoreDataGenerator;
 
     beforeEach(() => {
+        initialDataCreatorMock = Mock.ofInstance(() => null, MockBehavior.Strict);
+        assessments.forEach(assessment => {
+            (assessment as Assessment).initialDataCreator = initialDataCreatorMock.object;
+            initialDataCreatorMock.setup(mock => mock(assessment, null)).returns(() => null);
+        });
+        generator = new InitialAssessmentStoreDataGenerator(assessments);
         defaultState = generator.generateInitialState();
     });
 
     it('generates the pinned default state from assessmentsProvider data when no persistedData is provided', () => {
         expect(defaultState).toMatchSnapshot();
     });
 
-    it.each([[undefined], [null], [{}]])('outputs default assessment data if persistedData.assessments is %p', persistedAssessments => {
+    it.each([[undefined], [null]])('outputs default assessment data if persistedData.assessments is %p', persistedAssessments => {
+        const generatedState = generator.generateInitialState({
+            assessments: persistedAssessments,
+        } as IAssessmentStoreData);
+
+        expect(generatedState.assessments).toEqual(defaultState.assessments);
+    });
+
+    it('outputs default assessment data if persistedData.assessments is non-nullable', () => {
```

Comment:
>outputs default assessment data if persistedData.assessments is non-nullable [](start = 8, length = 76)

Can you improve this test's name? I don't think it's expressing a real requirement correctly (the name as written suggests that it should always be generating default data and never be using persisted data, even if present)

Reply:
Whoops, correct. It should be returning whatever data is given by initial data creator.